utility_network/perform-valve-isolation-trace/src/main/java/com/esri/samples/perform_valve_isolation_trace/PerformValveIsolationTraceController.java

```diff
@@ -0,0 +1,308 @@
+                  utilityNetwork.fetchFeaturesForElementsAsync(Collections.singletonList(startingLocation));
+
+          elementFeaturesFuture.addDoneListener(() -> {
+            try {
+              List<ArcGISFeature> startingLocationFeatures = elementFeaturesFuture.get();
+
+              if (!startingLocationFeatures.isEmpty()) {
+                Geometry startingLocationGeometry = startingLocationFeatures.get(0).getGeometry();
+
+                if (startingLocationGeometry instanceof Point){
+                  Point startingLocationGeometryPoint = (Point) startingLocationFeatures.get(0).getGeometry();
+
+                // create a graphic for the starting location and add it to the graphics overlay
+                Graphic startingLocationGraphic = new Graphic(startingLocationGeometry, startingPointSymbol);
+                startingLocationGraphicsOverlay.getGraphics().add(startingLocationGraphic);
+
+                // set the map's viewpoint to the starting location
+                mapView.setViewpointAsync(new Viewpoint(startingLocationGeometryPoint, 3000));
+
+                // build the choice list for categories populated with the 'Name' property of each 'UtilityCategory' in the 'UtilityNetworkDefinition'
+                categorySelectionComboBox.getItems().addAll(networkDefinition.getCategories());
+                categorySelectionComboBox.getSelectionModel().select(0);
+                categorySelectionComboBox.setCellFactory(param -> new UtilityCategoryListCell());
+                categorySelectionComboBox.setButtonCell(new UtilityCategoryListCell());
+
+                // enable the UI
+                enableUI();
+
+                // hide the progress indicator
+                progressIndicator.setVisible(false);
+
+                // update the status text
+                statusLabel.setText("Utility network loaded. Ready to perform trace...");
+                }
+
+              } else {
+                new Alert(Alert.AlertType.ERROR, "Error getting starting location geometry.").show();
+              }
+            } catch (ExecutionException | InterruptedException e) {
+              new Alert(Alert.AlertType.ERROR, "Error getting starting location feature.").show();
+            }
+          });
+
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Error loading Utility Network.").show();
+        }
+      });
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  @FXML
+  private void handleTraceClick() {
+    try {
+      // clear previous selection from the layers
+      mapView.getMap().getOperationalLayers().forEach(layer -> {
+        if (layer instanceof FeatureLayer) {
+          ((FeatureLayer) layer).clearSelection();
+        }
+      });
+
+      // show the progress indicator and update the status text
+      progressIndicator.setVisible(true);
+      statusLabel.setText("Running isolation trace...");
+
+      // disable the UI
+      traceButton.setDisable(true);
```

Comment:
with a slight tweak of the enableUI method to remove the progress indicator and take a boolean as a parameter, these lines could be neatened up with:
```suggestion
         `enableUI(true)`;
```
just a suggestion to consider since it would mean tweaking of code elsewhere: not a must!

Reply:
good idea, we can leave the progressIndicator in that method and just have it switch to the oposite of the param: `rogressIndicator.setVisible(!enable);`